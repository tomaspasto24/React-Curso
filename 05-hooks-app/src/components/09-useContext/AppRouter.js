import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
} from 'react-router-dom';
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
          <div>

              <NavBar />
            
              <div className='container'>
                <Routes>

                        {/* exact es para que sea la direcci√≥n exacta. 
                        element se ingresa el componente que tiene que ser renderizado. */}

                        <Route exact path='/' element={ <HomeScreen /> } />
                        <Route exact path='/about' element={ <AboutScreen /> } />
                        <Route exact path='/login' element={ <LoginScreen /> } />

                        {/* el path='*', se usa como por defecto. */}
                        <Route path='*' element={ <HomeScreen /> } />

                </Routes>
              </div>
          
          </div>  
        </Router>
    )
}
