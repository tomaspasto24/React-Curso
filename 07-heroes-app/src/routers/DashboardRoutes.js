import React from 'react';
import { Navbar } from '../components/ui/NavBar';
import { DCScreen } from '../components/dc/DcScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Routes, Route } from 'react-router-dom';
import { HeroScreen } from '../components/hero/HeroScreen';

export const DashboardRoutes = () => {
  return (
      <>
        <Navbar />

        <div className='container'>
          <Routes>
              <Route path="dc" element={ <DCScreen /> } />
              <Route path="marvel" element={ <MarvelScreen /> } />
              <Route path="search" element={ <SearchScreen /> } />
              {/* heroeId es un argumento obligatorio que se debe ingresar en el URL. */}
              <Route path="hero/:heroeId" element={ <HeroScreen /> } />
              <Route path="/" element={ <SearchScreen /> } />
          </Routes>
        </div>
      </>
  )
  ;
};
