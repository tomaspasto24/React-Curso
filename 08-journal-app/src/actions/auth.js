import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from "../types/types"
import { uiFinishLoading, uiStartLoading } from './ui';

export const startLoginEmailPassword = ( email, password ) => {
    
    // Retorna una func con parámetro dispatch cuando es una acción async.
    return ( dispatch ) => {

        dispatch( uiStartLoading() );

        firebase.auth().signInWithEmailAndPassword( email, password )
            .then( ({ user }) => {

                dispatch( login( user.uid, user.displayName ) );
                dispatch( uiFinishLoading() );

            })
            .catch( err => {

                console.log(err);
                dispatch( uiFinishLoading() );
            
            } );
        
    }
};

export const startRegisterWithEmailPasswordName = ( email, password, name) => {
    return ( dispatch ) => {
         
        firebase.auth().createUserWithEmailAndPassword( email, password ) 
            .then( async({ user }) => {

                // El displayName es null cuando se registra por email, asi que asignamos displayName con el name del usuario.
                await user.updateProfile({ displayName: name });

                dispatch( 
                    login( user.uid, user.displayName )
                )
            }) 
            .catch ( err => {
                console.log(err);
            });

    }
};

export const startGoogleLogin = () => {
    return ( dispatch ) => {
        
        // En caso de resolverse la petición, se hace dispatch (login) con el uid y displayName extraidos del usuario.
        firebase.auth().signInWithPopup( googleAuthProvider )
            .then( ( {user} ) => {
                dispatch( login( user.uid, user.displayName ) )
            } );
    
    };

};

export const login = ( uid, displayName ) => ({
    
    type: types.login,
    
    payload: {
        uid,
        displayName
    }

});