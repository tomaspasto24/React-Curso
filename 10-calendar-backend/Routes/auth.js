// Rutas

const { Router } = require("express");
const { check } = require('express-validator');

const { validarCampos } = require('../middlewares/validar-campos');

const router = Router();

const { crearUsuario, loginUsuario, revalidarToken } = require('../controllers/auth');

// /api/auth/new
router.post(
    '/new',
    [ // middlewares
        check('name', 'El nombre es obligatorio').not().isEmpty(),
        check('email', 'El email es obligatorio').isEmail(),
        check('password', 'El password es obligatorio, debe de ser de 6 caracteres').isLength({ min: 6 }),
        validarCampos
    ],
    crearUsuario );

router.post(
    '/',
    [ // middlewares
        check('email', 'El email es obligatorio').isEmail(),
        check('password', 'El password es obligatorio, debe de ser de 6 caracteres').isLength({ min: 6 }),
        validarCampos
    ],
    loginUsuario );

router.get( '/renew', revalidarToken );

module.exports = router;