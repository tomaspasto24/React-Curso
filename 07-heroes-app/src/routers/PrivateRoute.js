import React from 'react';
import { Navigate, useLocation } from 'react-router-dom';
import { useContext } from 'react/cjs/react.development';
import { AuthContext } from '../auth/authContext';

export const PrivateRoute = ({ children }) => {

    const { user } = useContext(AuthContext);
    const location = useLocation();

    localStorage.setItem('lastPath', location.pathname + location.search);

    return user.logged
        ? children
        : <Navigate to='/login' />
};
