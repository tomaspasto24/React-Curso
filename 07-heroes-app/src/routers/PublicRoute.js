import { useContext } from 'react/cjs/react.development';
import { AuthContext } from '../auth/authContext';
import { types } from '../types/types';

export const PublicRoute = ( { children } ) => {
    const {user, dispatch} = useContext(AuthContext);
    
    (user.logged) && dispatch({ type: types.logout })
    
    return children;
};
