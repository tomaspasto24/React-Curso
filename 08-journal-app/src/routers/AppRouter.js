import React, { useEffect } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { firebase } from '../firebase/firebase-config';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';

export const AppRouter = () => {

    const dispatch = useDispatch();
    
    // Es un efecto con una dependencia vacÃ­a por lo que solo se va a ejecutar una vez.(el dispatch es porque asi no queda como dependencia faltante).
    // Lo que hace es cargar el estado anterior, en caso de recargar el navegador. onAuthStateChanged siempre se va a estar ejecutando.
    useEffect( () => {
        firebase.auth().onAuthStateChanged( (user) => {

            if ( user?.uid ) {
                dispatch( login(user.uid, user.displayName) );
            }

        });
    }, [ dispatch ] );

    return (
        <Router>
            <div>
                <Switch>

                    <Route 
                        path='/auth' 
                        component={ AuthRouter } 
                    />
                    
                    <Route 
                        exact 
                        path='/' 
                        component={ JournalScreen } 
                    />

                    <Redirect
                        to='/auth/login'
                    />
                                    
                </Switch>
            </div>
        </Router>
    );
};
