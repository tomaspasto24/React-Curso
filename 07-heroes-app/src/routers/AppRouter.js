import React from 'react';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  return (
      <BrowserRouter>   
        <Routes>

            <Route path="/login" element={
              <PublicRoute>
                  <LoginScreen />
              </PublicRoute>
            } />
            
            {/* PrivateRoute se encarga de validar los datos de la variable user 
            para comrpobar si puede ver las rutas dentro de DashboardRoutes.
            Cuando se ingresa un componente dentro de otro, este termina ingresado en las props del componente padre.
            */}
            <Route path="/*" element={
              <PrivateRoute>
                 <DashboardRoutes />
              </PrivateRoute>
            } />

        </Routes>
      </BrowserRouter>
  );
};
