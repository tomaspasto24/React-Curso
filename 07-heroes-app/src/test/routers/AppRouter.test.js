import { AppRouter } from "../../routers/AppRouter";
import { mount } from 'enzyme';
import { AuthContext } from "../../auth/authContext";

describe('Pruebas sobre AppRouter', () => {
    
    const contextValue = {
        user: {
            logged: false
        }
    };

    test('Debe de mostrar el login si no esta autenticado el usuario.', () => {
        
        const wrapper = mount( <AuthContext.Provider value={ contextValue }>
            <AppRouter/>
        </AuthContext.Provider>
        );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe('Login Screen');
    });

});