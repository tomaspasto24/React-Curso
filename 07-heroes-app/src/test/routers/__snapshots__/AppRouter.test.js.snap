// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre AppRouter Debe de mostrar el login si no esta autenticado el usuario. 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
